tracer/src/Datadog.Trace/ISpanContextExtractor.cs

```diff
@@ -19,10 +19,7 @@ namespace Datadog.Trace
     public interface ISpanContextExtractor
     {
         /// <summary>
-        /// Given a SpanContext carrier and a function to access the values, this method will extract SpanContext if any
-        /// When enabled (and present in the headers) a data streams monitoring checkpoint is set.
-        /// You should only call <see cref="Extract{TCarrier}"/> once on the message <paramref name="carrier"/>. Calling
-        /// multiple times may lead to incorrect stats when using Data Streams Monitoring.
```

Comment:
This still _sort_ of applies, right? We don't set a checkpoint, but customers _should_ still call it to correctly align their checkpoints?

Reply:
hmm, yes, sure, they can/should call this to appropriately pair messages checkpoints, but all the statements here are false with the new code:
 - no checkpoint is set
 - the method can be called several times
 - it will not lead to incorrect data in DSM

Reply:
maybe I can add something explaining how this method can be used in the context of DSM instead ?

Reply:
> maybe I can add something explaining how this method can be used in the context of DSM instead ?

Might be useful for DSM customers? But I'm happy to defer to you on that 🙂

Reply:
it's a pretty specific corner case that's useful only for kafka, and when DD_TRACE_KAFKA_CREATE_CONSUMER_SCOPE_ENABLED is disabled, so I can explain that in the comment, but it might be a bit verbose

Reply:
I mean, most users are using this method only for the trace propagation I suppose.